src/contexts/AppProvider.tsx

```diff
@@ -13,9 +15,12 @@ export default function AppProvider({ children }: Props) {
     <Boundary>
       <ToastProvider>
         <AuthProvider>
-          <SwrProvider>
-            <ModalProvider>{children}</ModalProvider>
-          </SwrProvider>
+          <Web3ProviderNetwork>
```

Comment:
this is the recommended way to maintain another web3 connection - this way we can have a web3 connection to read contract data to render the contract details on a page without a user connecting their wallet,, and a separate connection to do user wallet stuff
https://github.com/NoahZinsmeister/web3-react/tree/v6/docs#createweb3reactroot